ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java

```diff
@@ -545,7 +545,14 @@ public void setUnderlineColor(ReactEditText view, @Nullable Integer underlineCol
     if (underlineColor == null) {
       drawableToMutate.clearColorFilter();
     } else {
-      drawableToMutate.setColorFilter(underlineColor, PorterDuff.Mode.SRC_IN);
+      if (Build.VERSION.SDK_INT == Build.VERSION_CODES.LOLLIPOP) {
+        int bottomBorderColor = view.getBorderColor(Spacing.BOTTOM);
+        setBorderColor(view, Spacing.START, underlineColor);
```

Comment:
Still not clear to me why we are using `Spacing.START`. Understanding it and testing it in a scenario/test case that uses `Spacing.END` may cause some issues.. I'll work on this next week.